linker/src/lib.rs

```diff
@@ -236,29 +257,43 @@ fn process_link(link: Link) -> PilStatement {
             .map(|oid| namespaced_reference(to_namespace.clone(), oid))
             .into_iter();
 
-        // plookup rhs is `latch $ [ operation_id, inputs, outputs ]`
-        let latch = Some(namespaced_reference(
+        let latch = Box::new(namespaced_reference(
             to_namespace.clone(),
             to.machine.latch.unwrap(),
         ));
 
-        let rhs = SelectedExpressions {
-            selector: latch,
-            expressions: Box::new(
-                ArrayLiteral {
-                    items: op_id
-                        .chain(to.operation.params.inputs_and_outputs().map(|i| {
-                            index_access(
-                                namespaced_reference(to_namespace.clone(), &i.name),
-                                i.index.clone(),
-                            )
-                        }))
-                        .collect(),
-                }
-                .into(),
-            ),
-        };
-        PilStatement::PlookupIdentity(SourceRef::unknown(), lhs, rhs)
+        // plookup rhs is `latch $ [ operation_id, inputs, outputs ]`
+        let rhs = Expression::BinaryOperation(
+            SourceRef::unknown(),
+            BinaryOperation {
+                left: latch,
+                op: BinaryOperator::Select,
+                right: Box::new(
+                    ArrayLiteral {
+                        items: op_id
+                            .chain(to.operation.params.inputs_and_outputs().map(|i| {
+                                index_access(
+                                    namespaced_reference(to_namespace.clone(), &i.name),
+                                    i.index.clone(),
+                                )
+                            }))
+                            .collect(),
+                    }
+                    .into(),
+                ),
+            },
+        );
+
+        let identity_expr = Expression::BinaryOperation(
```

Comment:
Maybe in general, it might be useful to create helper functions to create lookups. Maybe in the same file as the `namespaced_reference` function?